pkg/workflowengine/impl/propeller_executor.go

```diff
@@ -3,10 +3,12 @@ package impl
 import (
```

Comment:
Something is a bit confusing about the structure of the code here... 
why is this called propeller_executor?

It feels as if we have sort of two abstractions, plz correct me if this is wrong,
1. workflow engine abstracts the execution of workflows... with "propeller" as "one" possible executor.. but no way to select another
2. Within propeller, it abstracts the execution of workflows on k8s and we have one default implementation that creates the WF CRD in the target cluster directly and we are introducing a way to register separate k8s executors... right?

If that's the case, I would like to understand why? What does the propeller abstraction add that can't be moved up one level to the workflow engine abstraction (metrics?) ? and why are we only opening a way to registering separate k8s executors but not replace propeller executor?